lib/webauthn/public_key_credential.rb

```diff
@@ -1,13 +1,46 @@
 # frozen_string_literal: true
 
 require "base64"
+require "webauthn/authenticator_assertion_response"
+require "webauthn/authenticator_attestation_response"
 
 module WebAuthn
   class PublicKeyCredential
     VALID_TYPE = "public-key"
 
     attr_reader :type, :id, :raw_id, :response
 
+    def self.from_create(credential)
+      new(
+        type: credential["type"],
+        id: credential["id"],
+        raw_id: decode(credential["rawId"]),
+        response: WebAuthn::AuthenticatorAttestationResponse.new(
+          attestation_object: decode(credential["response"]["attestationObject"]),
+          client_data_json: decode(credential["response"]["clientDataJSON"])
+        )
+      )
+    end
+
+    def self.from_get(credential)
+      new(
+        type: credential["type"],
+        id: credential["id"],
+        raw_id: decode(credential["rawId"]),
+        response: WebAuthn::AuthenticatorAssertionResponse.new(
+          # FIXME: credential_id doesn't belong inside AuthenticatorAssertionResponse
+          credential_id: decode(credential["id"]),
+          authenticator_data: decode(credential["response"]["authenticatorData"]),
+          client_data_json: decode(credential["response"]["clientDataJSON"]),
+          signature: decode(credential["response"]["signature"])
+        )
+      )
+    end
+
+    def self.decode(data)
+      Base64.urlsafe_decode64(data)
```

Comment:
For now this is urlsafe-base64 because it's the one used by the FIDO2 conformance tools. But the plan is to make it configurable.

Reply:
Added support for both base64 and base64url, with base64 as the default.

Reply:
Changed my mind. I think we should have `base64url` as the default, so it works out of the box with FIDO2 tools and with `webauthn-json` (see https://github.com/github/webauthn-json/issues/2).